### PR TITLE
[MI-2056]: Fixed create task flow and added user connection check on …

### PR DESCRIPTION
**Pull request by avas27JTG**
Fixed create task flow and added user connection check on reload

### CODE REVIEW

**Review by Kshitij-Katiyar — APPROVED** [review]

**Review by sooraj-shukla — APPROVED** [review]

File: webapp/src/selectors/index.tsx [hunk 1]
```diff
@@ -16,6 +16,10 @@ export const getLinkModalState = (state: any): LinkProjectModalState => {
     return state[pluginPrefix].openLinkModalSlice;
 };
 
+export const getCreateTaskModalState = (state: any): CreateTaskModalState => {
```
Comment (manojmalik20) [inline webapp/src/selectors/index.tsx:19]: Change the type of `state` from any to the correct type.

File: webapp/src/types/common/store.d.ts [hunk 2]
```diff
@@ -14,3 +14,13 @@ type LinkProjectModalState = {
     project: string,
     isLinked: boolean,
 }
+
+type TaskFieldsCommandArgs = {
```
Comment (manojmalik20) [inline webapp/src/types/common/store.d.ts:18]:
```suggestion
type CreateTaskCommandArgs = {
```

File: webapp/src/components/modal/index.tsx [hunk 3]
```diff
@@ -25,7 +25,7 @@ type ModalProps = {
     cancelDisabled?: boolean;
 }
 
-const Modal = ({show, onHide, showCloseIconInHeader = true, children, title, subTitle, onConfirm, confirmAction, confirmBtnText, cancelBtnText, className = '', loading = false, error}: ModalProps) => {
+const Modal = ({show, onHide, showCloseIconInHeader = true, children, title, subTitle, onConfirm, confirmAction, confirmBtnText, cancelBtnText, confirmDisabled, className = '', loading = false, error}: ModalProps) => {
     return (
```
[the remark this answers is not in the document]
Comment (avas27JTG) [reply webapp/src/components/modal/index.tsx:29]: these are the components copied from Sooraj's changes on SN, we will need to fix there as well or can clean at once when using UI lib.

however fixed here

File: webapp/src/containers/TaskModal/index.tsx [hunk 4]
```diff
@@ -56,175 +58,198 @@ const taskTypeOptions = [
...
-            const {isSuccess, isError} = usePlugin.getApiState(Constants.pluginApiServiceConfigs.createTask.apiServiceName, taskPayload);
-            if ((isSuccess && !isError) || (!isSuccess && isError)) {
-                onHide();
+        if (getCreateTaskModalState(usePlugin.state).visibility) {
+            // Pre-select the dropdown value in case of single option.
+            if (organizationOptions.length === 1) {
+                setTaskDetails({...taskDetails, organization: organizationOptions[0].value});
             }
+            if (projectOptions.length === 1) {
+                setTaskDetails({...taskDetails, project: projectOptions[0].value});
```
[the remark this answers is not in the document]
Comment (avas27JTG) [reply webapp/src/containers/TaskModal/index.tsx:182]: all these changes are completely reverted in later PR https://github.com/Brightscout/mattermost-plugin-azure-devops/pull/61
so, not making any changes here

File: webapp/src/selectors/index.tsx (hunk 1 above)
Comment (avas27JTG) [reply webapp/src/selectors/index.tsx:19]: already fixed

File: webapp/src/containers/TaskModal/index.tsx [hunk 5]
```diff
@@ -56,175 +58,198 @@ const taskTypeOptions = [
...
+        }
+    };
 
+    // Set modal field values
     useEffect(() => {
-        if (taskPayload) {
-            const {isSuccess, isError} = usePlugin.getApiState(Constants.pluginApiServiceConfigs.createTask.apiServiceName, taskPayload);
-            if ((isSuccess && !isError) || (!isSuccess && isError)) {
-                onHide();
+        if (getCreateTaskModalState(usePlugin.state).visibility) {
```
[the remark this answers is not in the document]
Comment (avas27JTG) [reply webapp/src/containers/TaskModal/index.tsx:176]: this has been refactored in later PRs

**Review by manojmalik20 — APPROVED** [review]

**Pull request merged**